internal/operators/localmanager/localmanager.go

```diff
@@ -0,0 +1,343 @@
+		case runtimeclient.ContainerdName:
+			// socketPath = commonFlags.RuntimesSocketPathConfig.Containerd
+		case runtimeclient.CrioName:
+			// socketPath = commonFlags.RuntimesSocketPathConfig.Crio
+		default:
+			return commonutils.WrapInErrInvalidArg("--runtime / -r",
+				fmt.Errorf("runtime %q is not supported", p))
+		}
+
+		for _, r := range rc {
+			if r.Name == runtimeName {
+				log.Infof("Ignoring duplicated runtime %q from %v",
+					runtimeName, parts)
+				continue partsLoop
+			}
+		}
+
+		rc = append(rc, &containerutils.RuntimeConfig{
+			Name:       runtimeName,
+			SocketPath: socketPath,
+		})
+	}
+
+	l.rc = rc
+
+	localGadgetManager, err := localgadgetmanager.NewManager(l.rc)
+	if err != nil {
+		return commonutils.WrapInErrManagerInit(err)
+	}
+	l.localGadgetManager = localGadgetManager
+	return nil
+}
+
+func (l *LocalManager) Close() error {
+	l.localGadgetManager.Close()
+	return nil
+}
+
+func (l *LocalManager) Instantiate(runner operators.Runner, tracer any, perGadgetParams *params.Params) (operators.OperatorInstance, error) {
+	_, canEnrichEvent := runner.Gadget().EventPrototype().(operators.ContainerInfoFromMountNSID)
+
+	traceInstance := &localManagerTrace{
+		LocalManager:       l,
+		enrichEvents:       canEnrichEvent,
+		attachedContainers: make(map[*containercollection.Container]struct{}),
+		perGadgetParams:    perGadgetParams,
+		tracer:             tracer,
+		runner:             runner,
+	}
+
+	return traceInstance, nil
+}
+
+type localManagerTrace struct {
+	*LocalManager
+	mountnsmap      *ebpf.Map
+	enrichEvents    bool
+	subscriptionKey string
+
+	// Keep a map to attached containers, so we can clean up properly
+	attachedContainers map[*containercollection.Container]struct{}
+	attacher           Attacher
+	perGadgetParams    *params.Params
+	tracer             any
+	runner             operators.Runner
+}
+
+func (l *localManagerTrace) Name() string {
+	return OperatorInstanceName
+}
+
+func (l *localManagerTrace) PreGadgetRun() error {
+	log := l.runner.Logger()
+
+	// TODO: Improve filtering, see further details in
+	// https://github.com/inspektor-gadget/inspektor-gadget/issues/644.
+	containerSelector := containercollection.ContainerSelector{
+		Name: l.perGadgetParams.Get(ContainerName).AsString(),
+	}
+
+	if setter, ok := l.tracer.(MountNsMapSetter); ok {
+		// Create mount namespace map to filter by containers
+		mountnsmap, err := l.localGadgetManager.CreateMountNsMap(containerSelector)
+		if err != nil {
+			return commonutils.WrapInErrManagerCreateMountNsMap(err)
+		}
+
+		log.Debugf("set mountnsmap for gadget")
+		setter.SetMountNsMap(mountnsmap)
+
+		l.mountnsmap = mountnsmap
+	}
+
+	if setter, ok := l.tracer.(ContainersMapSetter); ok {
+		setter.SetContainersMap(l.localGadgetManager.ContainersMap())
+	}
+
+	if attacher, ok := l.tracer.(Attacher); ok {
+		l.attacher = attacher
+
+		attachContainerFunc := func(container *containercollection.Container) {
+			var cbFunc any
```

Comment:
Just trying to understand this, don't we need to assign something to `cbFunc` or why do have it here?

Reply:
Ah, this is a leftover, I need to remove that - callbacks can be removed, because AttachGeneric gives all information needed to enrich as "container" anyways - the gadget can simply use that.